src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        Node<K, V> listNode = table[index];
+        while (true) {
```

Comment:
```suggestion
        if (listNode == null) {
            // add node and call return;
        }
        while (listNode.next != null || Objects.equals(key, listNode.key) {
             // here iterate your singly linked list
        }
```
that's approximately what I meant